tests/nested/manual/uc-update-assets-secure-add-sbat/task.yaml

```diff
@@ -0,0 +1,154 @@
+  # Save that file
+  cp pc/grubx64.efi grubx64.efi
+
+  cp old-shimx64.efi pc/shim.efi.signed
+
+  # Repack pc gadget for the initial image
+  tests.nested secboot-sign file pc/shim.efi.signed "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  tests.nested secboot-sign file pc/grubx64.efi "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  old_shim_sha="$(sha256sum pc/shim.efi.signed | sed "s/ .*//")"
+  old_grub_sha="$(sha256sum pc/grubx64.efi | sed "s/ .*//")"
+  snap pack pc "$(tests.nested get extra-snaps-path)"
+
+  tests.nested build-image core
+  tests.nested create-vm core
+
+  cat <<EOF >sbat.csv
+  sbat,1,SBAT Version,sbat,1,https://github.com/rhboot/shim/blob/main/SBAT.md
+  grub,1,Free Software Foundation,grub,2.06,https://www.gnu.org/software/grub/
+  grub.ubuntu,1,Canonical,grub2,2.06-2ubuntu7,https://packages.ubuntu.com/source/jammy/grub2
+  EOF
+  sbatsize="$(($(stat -c %b sbat.csv)*$(stat -c %B sbat.csv)))"
+  truncate --size "${sbatsize}" sbat.csv
+
+  # Add .sbat at the end
+  grubsize="$(($(stat -c %b grubx64.efi)*$(stat -c %B grubx64.efi)))"
+  objcopy --add-section .sbat=sbat.csv --change-section-address .sbat="${grubsize}" grubx64.efi grubx64-sbat.efi
+
+  cp new-shimx64.efi pc/shim.efi.signed
+  cp grubx64-sbat.efi pc/grubx64.efi
+
+  # Sign modified grub and shim
+  tests.nested secboot-sign file pc/shim.efi.signed "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  tests.nested secboot-sign file pc/grubx64.efi "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  new_shim_sha="$(sha256sum pc/shim.efi.signed | sed "s/ .*//")"
+  new_grub_sha="$(sha256sum pc/grubx64.efi | sed "s/ .*//")"
+
+  if [ "${UPDATE_SEED}" = true ]; then
+    (cd pc && python3 "${TESTSLIB}/bump_asset_edition.py" --role system-seed)
+  fi
+  if [ "${UPDATE_BOOT}" = true ]; then
+    (cd pc && python3 "${TESTSLIB}/bump_asset_edition.py" --role system-boot)
+  fi
+
+  cat <<EOF >>expected-before
+  ${old_shim_sha} */boot/efi/EFI/boot/bootx64.efi
+  ${old_grub_sha} */boot/efi/EFI/boot/grubx64.efi
+  EOF
+  if [ "${UPDATE_SEED}" = true ]; then
+    cat <<EOF >>expected-after
+  ${new_shim_sha} */boot/efi/EFI/boot/bootx64.efi
+  ${new_grub_sha} */boot/efi/EFI/boot/grubx64.efi
+  EOF
+  else
+    cat <<EOF >>expected-after
+  ${old_shim_sha} */boot/efi/EFI/boot/bootx64.efi
+  ${old_grub_sha} */boot/efi/EFI/boot/grubx64.efi
+  EOF
+  fi
+
+  cat <<EOF >>expected-before
+  ${old_shim_sha} */run/mnt/ubuntu-boot/EFI/boot/bootx64.efi
+  ${old_grub_sha} */run/mnt/ubuntu-boot/EFI/boot/grubx64.efi
+  EOF
+  if [ "${UPDATE_BOOT}" = true ]; then
+    cat <<EOF >>expected-after
+  ${new_shim_sha} */run/mnt/ubuntu-boot/EFI/boot/bootx64.efi
+  ${new_grub_sha} */run/mnt/ubuntu-boot/EFI/boot/grubx64.efi
+  EOF
+  else
+    cat <<EOF >>expected-after
+  ${old_shim_sha} */run/mnt/ubuntu-boot/EFI/boot/bootx64.efi
+  ${old_grub_sha} */run/mnt/ubuntu-boot/EFI/boot/grubx64.efi
+  EOF
+  fi
+
+  snap pack pc --filename=pc_2.snap
+
+  tests.nested exec systemctl --wait is-system-running || true
```

Comment:
you need to use remote.exec instead of tests.nested exec
This was changed recently.

Reply:
this applies to all the tests

Reply:
Done